cluster-up/cluster/kind-ovn/provider.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash -ex
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2024 Red Hat, Inc.
+#
+
+KIND_VERSION=0.19.0
+export KIND_IMAGE=kindest/node
+export K8S_VERSION=v1.28.0@sha256:dad5a6238c5e41d7cac405fae3b5eda2ad1de6f1190fa8bfc64ff5bb86173213
+
+KIND_PATH=${KIND_PATH:-"${KUBEVIRTCI_CONFIG_PATH}/${KUBEVIRT_PROVIDER}/_kind"}
+CLUSTER_PATH=${CLUSTER_PATH:-"${KUBEVIRTCI_CONFIG_PATH}/${KUBEVIRT_PROVIDER}/_ovnk"}
+
+MTU=${MTU:-"1392"}
+
+PLATFORM=$(uname -m)
+case ${PLATFORM} in
+x86_64* | i?86_64* | amd64*)
+    ARCH="amd64"
+    ;;
+ppc64le)
+    ARCH="ppc64le"
+    ;;
+aarch64* | arm64*)
+    ARCH="arm64"
+    ;;
+*)
+    echo "invalid Arch, only support x86_64, ppc64le, aarch64"
+    exit 1
+    ;;
+esac
+
+function fetch_kind() {
+    mkdir -p $KIND_PATH
+    current_kind_version=$($KIND_PATH/kind --version |& awk '{print $3}')
+    if [[ $current_kind_version != $KIND_VERSION ]]; then
+        echo "Downloading kind v$KIND_VERSION"
+        curl -LSs https://github.com/kubernetes-sigs/kind/releases/download/v$KIND_VERSION/kind-linux-${ARCH} -o "$KIND_PATH/kind"
+        chmod +x "$KIND_PATH/kind"
+    fi
+    export PATH=$KIND_PATH:$PATH
+}
+
+function prepare_config() {
+    echo "STEP: Prepare provider config"
+    cat >$KUBEVIRTCI_CONFIG_PATH/$KUBEVIRT_PROVIDER/config-provider-$KUBEVIRT_PROVIDER.sh <<EOF
+master_ip=127.0.0.1
+kubeconfig=${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig
+kubectl=kubectl
+docker_prefix=127.0.0.1:5000/kubevirt
+manifest_docker_prefix=localhost:5000/kubevirt
+EOF
+}
+
+function up() {
+    cluster::install
+    fetch_kind
+
+    if [[ ${CI} == "true" ]]; then
+        sysctl fs.inotify.max_user_watches=1048576
+        sysctl fs.inotify.max_user_instances=512
+        dnf install -y openssl
+    fi
```

Comment:
I remember needing this at my laptop too, not only CI.

Reply:
user would just need to do it once manually imo, we can add a readme
(locally i didnt but this is just a specific machine)

Reply:
Configuring the max user instances is something I have to do locally (I use podman).
Not sure about the max user watches.

Reply:
> user would just need to do it once manually imo, we can add a readme, wdyt ? (locally i didnt but this is just a specific machine)
> 
> btw maybe worth without the watches on CI ? or you saw you must have them on CI ?

This is idempotent configuration, not big deal with doing it every time, let's just remove the CI env var and that's it.

Reply:
note this is not something we are doing on any other provider,
i mean if some operations like installing stuff, config (for example disable_ipv6=0) etc
we require the user to do it just once, i think it is wrong and might require root (maybe we need root anyhow)
but if you insist i will do that
note that soon we will have a CI image that have 3 lines already, so it will be required only locally
which is even worse

Reply:
I really don't follow your argument @oshoval

Reply:
ok tested with the new CI image `quay.io/kubevirtci/golang-legacy:v20240523-1a2c9b8`
we don't need on CI those lines anymore because now they are part of the image
```
        sysctl fs.inotify.max_user_watches=1048576
        sysctl fs.inotify.max_user_instances=512
        dnf install -y openssl
```
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3426/rehearsal-pull-kubevirt-e2e-net-integration-tests/1793555893502087168

so now we don't need those on CI, lets talk about locally
until today, all the providers we created / have, don't install packages that are needed on the root level
(referring to openssl that was required), it is like if the provider need `wget` it won't install, the user need to.
and they also don't set configuration, for example disable_ipv6 must be 0, and it is the user responsibility to set it, the provider doesn't do it it each time

So please reconsider, do you still want to include those lines that now only required locally,
or to just add a readme that say what is needed (or even just add it on the PR desc) ?
fwiw we don't get into resolution in the readme of what should be dnf installed, and even about some of the configuration (such as disable_ipv6=0) we don't usually.

Reply:
I really don't care. 

As I see it, the 2 options are:
- hand a command to the user that just works. downside is ?... I don't see one.
- hand a command to the user that depends on the configuration of their environment. If / once the commands fails on their environment, the user will need to understand what's happening / look for a readme, and understand the dependencies, then act upon them

If the kubevirtci way is to do the latter, by all means, apply that. I know what I would rather have as a user.